README.md

```diff
@@ -293,6 +294,42 @@ fastify.post('/upload/files', {
 })
 ```
 
+If provided, the `sharedSchemaId` parameter must be a string ID and a shared schema will be added to your fastify instance so you will be able to apply the validation to your service (like in the example mentioned above).
+
+The shared schema, that is added, will look like this:
+```js
+{
+  type: 'object',
+  properties: {
+    encoding: { type: 'string' },
+    filename: { type: 'string' },
+    limit: { type: 'boolean' },
+    mimetype: { type: 'string' }
+  }
+}
+```
+**Note**:
+When sending fields with the body (`attachFieldsToBody` set to true), a field might look like this:
+```json
+{
+  "hello": "world"
+}
+```
+The mentioned field will be converted, by the plugin, to a more complex field under the hood. The converted field will look something like this:
```

Comment:
This is not totally clear to me reading the docs what is the trigger that transform the `hello` field into a complex one

If `attachFieldsToBody:true` we will get `request.body` as `{ "hello": "world"}`
If "something else" we will get the `request.body.hello` as a complex field

Reply:
Actually the plugin will ALWAYS convert ALL the non file fields.

Reply:
Ok, so we are saying that the `hello: world` is the expectation, but the user will get the complex one.
I would remove this example, it carries the attention to what the user will not have in its hands.

I would just say:

> Plain fields' JSON structure is like: ..

And I would put a JSON schema near the example, this would be really helpful for users

Reply:
Well, the whole reason of bringing this to the users attention is that when adding a the JSON schema for the plain field would be expected to be:
```js
hello: {
  type: 'string'
}
```
But instead (because of the conversion that is NOT mentioned anywhere) the user has to write it like this:
```js
hello: {
  properties: {
    value: { 
      type: 'string',
    }
  }
 }
```
If we don't mention the conversion and the result of it - regardless of the fact that it's not in the users hands, then the schema will not make sense or it will be confusing (like it was for me before).

Reply:
I've added the JSON schema bit under the example.